simphony/io/h5_cuds.py

```diff
@@ -208,6 +262,21 @@ def delete_mesh(self, name):
             raise ValueError(
                 'Mesh \'{n}\` does not exist'.format(n=name))
 
+    def delete_lattice(self, name):
+        """Delete lattice from file.
+
+        Parameters
+        ----------
+        name : str
+            name of lattice to delete
+        """
+        try:
+            filelat = self._handle.root.lattice._f_get_child(name)
+            filelat._f_remove(recursive=True)
```

Comment:
I would move this line out of the try... except block, it is prefered to have only the code that would raise an exception in the try block. Consider the alternative
```python
try:
    filelat = self._handle.root.lattice._f_get_child(name)
except tables.NoSuchNodeError:
    raise ValueError('Lattice \'{n}\` does not exist'.format(n=name))
else:
    filelat._f_remove(recursive=True)
```